### ISSUE TITLE
Nunjucks will only output some properties to the page

### ISSUE THREAD

**Issue by Jermaine0Forbes** [mozilla/nunjucks#1007]
Hello,  I have this weird situation that some properties will be printed out to the page when I get data from MongoDB. I highly doubt that mongodb is the problem because when I used mysql  I had similar problems .  Whenever I try to output some properties to web page only a couple show up.

In this image there is a number of properties I want to display
![step1](https://user-images.githubusercontent.com/20468655/27366057-b5c09b70-5611-11e7-879b-fd8fb47d0752.jpg)

But, when you look at the web page some of the properties are  displayed on the page except  the specific ones that I wrapped in a red box
![step2](https://user-images.githubusercontent.com/20468655/27366099-f601d53c-5611-11e7-87f6-bc02b8b7ded3.jpg)

Here is proof that the data is being passed from mongoose to the page because I made sure to console log the information
![step3](https://user-images.githubusercontent.com/20468655/27366127-20b5f3a8-5612-11e7-828a-21b39d954898.jpg)

And the weird part is that when I output  the object itself it will  show all the properties, including the ones that are not displaying by themselves. 
![step4](https://user-images.githubusercontent.com/20468655/27366181-6a7a484a-5612-11e7-8da5-a41c1fb49b35.jpg)

This is the proof that it will display  all of the properties if I just output the object
![step5](https://user-images.githubusercontent.com/20468655/27366206-94da2efc-5612-11e7-865c-b1cd8725fc5e.jpg)

**Comment by fdintino** [comment]
What does the output html markup look like? Are the elements there? Is it possible that they're not showing up because they are being hidden by css or are outside of an overflow hidden area?

**Comment by Jermaine0Forbes** [comment]
No that cannot be the case, if that was the case then you would not see all the properties when I just displayed the object. There is no overflow hidden outside of the summary text. Also,  I had this problem before with a different node application I was working on and I didn't have any  css styling to the other one

**Comment by fdintino** [comment]
It would be very helpful to see the html output from view source. It's impossible to troubleshoot based on a screenshot.

**Comment by fdintino** [comment]
Another detail that would be relevant is what library you are using to pull the data from mongo.

**Comment by Jermaine0Forbes** [comment]
Here you go pushed it into github [here is the link](https://github.com/Jermaine0Forbes/FitnessWebsite/blob/master/mvc/views/home.html). Also, I was using mongoose to retrieve the data

**Comment by Jermaine0Forbes** [comment]
Also it is line 102 you should look for. I forgot to tell you that

**Comment by ArmorDarks** [comment]
Hm, only I've noticed some strange thing about `{{ d }}` example? It had to output `[object Object]` instead of printing full JSON. Or did you use it like `{{ d|dump() }}`?

Though, maybe it's Nunjucks 3.x specific behavior. We're still on 2.x.

**Comment by fdintino** [comment]
Looking at your mongoose schema ([this file](https://github.com/Jermaine0Forbes/FitnessWebsite/blob/master/mvc/models/schema.js)) it looks like you haven't specified month and year. When you use `{{ d }}` in the template, that is effectively calling `d.toString()`, which I'm guessing will serialize the entire document, without regard to the schema. But if you try to access a property individually and it is not on the schema, the behavior must be different in a way that is causing this issue.

Try adding slug, month, and year to your blog schema. I suspect that will fix the issue on your end.

**Comment by Jermaine0Forbes** [comment]
Hello, to answer fdintino comment, the schema file that you are looking at it  is from an old branch called "fit1". That branch is the first one you will see when going through the FitnessWebsite repository. The current branch I am working on is "master".  In any case, I am going to close this issue because I just used a different templating engine(EJS) and I had the same problem so I guess it is not  nunjucks. I did resolve the issue through EJS  by using the method " toObject(); " on the object before printing out the properties. This method made data the show. I don't think nunjucks has something equivalent to that. I don't know if it is a MongoDB or mongoose issue, but now I am finding out that it is not a nunjucks issue
